dns-query/src/adapter.ts

```diff
@@ -0,0 +1,59 @@
+import { DNSResponse, Execute } from '@chainlink/types'
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { util } from '@chainlink/ea-bootstrap'
+
+export enum DNSProviders {
+  Cloudfare = 'cloudfare',
+  Google = 'google',
+}
+
+export const endpoints: Record<string, string> = {
+  [DNSProviders.Cloudfare]: 'https://cloudflare-dns.com/dns-query',
+  [DNSProviders.Google]: 'https://dns.google/dns-query',
+}
+
+export const inputParams = {
+  name: true,
+  type: true,
+  do: false,
+  cd: false,
+}
+
+export const execute: Execute = async (input) => {
+  try {
+    const validator = new Validator(input, inputParams)
+    if (validator.error) throw validator.error
+
+    const dnsProvider = util.getRequiredEnv('DNS_PROVIDER')
+    if (!Object.values(DNSProviders).includes(dnsProvider))
+      throw new Error(`Unknown DNS Provider: ${dnsProvider}`)
+
+    const jobRunID = validator.validated.id
+    const { name, type, do: doBit, cd: cdBit } = validator.validated.data
+
+    const params = {
+      name,
+      type,
+      ...(doBit && { do: doBit }),
+      ...(cdBit && { cd: cdBit }),
+    }
+    const headers = {
+      Accept: 'application/dns-json',
+    }
+
+    const result = await Requester.request({
+      url: endpoints[dnsProvider],
+      headers,
+      params,
+    })
+
+    const data: DNSResponse = { ...result.data }
+
+    return Requester.success(jobRunID, {
+      status: 200,
+      data: data,
+    })
+  } catch (e) {
+    console.error(e)
+  }
```

Comment:
Is this leftover from debugging? No need to catch any exceptions here, just pass them through.
```suggestion
```

Reply:
It was for debugging. But note that the adapter doesn't log any error right now

Reply:
It's returned to the caller but not logged? We can introduce error logging in the `bootstrap` module if that is the case.

Reply:
We can open an issue on it

Reply:
Yeah, please open an issue so we can keep track of it. Thanks!

Reply:
[Log adapter errors #204](https://github.com/smartcontractkit/external-adapters-js/issues/204) issue is opened to track this.